src/com/cloudogu/ces/cesbuildlib/Trivy.groovy

```diff
@@ -0,0 +1,133 @@
+     * @param additionalFlags Additional Trivy command flags
+     * @param trivyReportFile Location of Trivy report file. Should be set individually when scanning multiple images in the same pipeline
+     * @return Returns true if the scan was ok (no vulnerability found); returns false if any vulnerability was found
+     */
+    boolean scanDogu(
+        String doguDir = ".",
+        String severityLevel = TrivySeverityLevel.CRITICAL,
+        String strategy = TrivyScanStrategy.UNSTABLE,
+        // Avoid rate limits of default Trivy database source
+        String additionalFlags = "--db-repository public.ecr.aws/aquasecurity/trivy-db --java-db-repository public.ecr.aws/aquasecurity/trivy-java-db",
+        String trivyReportFile = "trivy/trivyReport.json"
+    ) {
+        String image = script.sh(script: "jq .Image ${doguDir}/dogu.json", returnStdout: true).trim()
+        String version = script.sh(script: "jq .Version ${doguDir}/dogu.json", returnStdout: true).trim()
+        return scanImage(image+":"+version, severityLevel, strategy, additionalFlags, trivyReportFile)
+    }
+
+    /**
+     * Save the Trivy scan results as a file with a specific format
+     *
+     * @param format The format of the output file (@see TrivyScanFormat)
+     * @param formattedTrivyReportFilename The file name your report files should get, without file extension. E.g. "ubuntu24report"
+     * @param trivyReportFile The "trivyReportFile" parameter you used in the "scanImage" function, if it was set
+     */
+    void saveFormattedTrivyReport(String format = TrivyScanFormat.HTML, String formattedTrivyReportFilename = "trivyReport", String trivyReportFile = "trivy/trivyReport.json") {
+        String fileExtension
+        String formatString
+        String trivyDirectory = "trivy/"
+        switch (format) {
+            case TrivyScanFormat.HTML:
+                formatString = "template --template \"@/contrib/html.tpl\""
+                fileExtension = "html"
+                break
+            case TrivyScanFormat.JSON:
+                formatString = "json"
+                fileExtension = "json"
+                break
+            case TrivyScanFormat.TABLE:
+                formatString = "table"
+                fileExtension = "txt"
+                break
+            default:
+                script.error("This format did not match the supported formats: " + format)
+                return
+        }
+        docker.image("aquasec/trivy:${trivyVersion}")
+            .inside("-v ${script.env.WORKSPACE}/.trivy/.cache:/root/.cache/") {
+                script.sh(script: "trivy convert --format ${formatString} --output ${trivyDirectory}${formattedTrivyReportFilename}.${fileExtension} ${trivyReportFile}")
+            }
+        script.archiveArtifacts artifacts: "${trivyDirectory}${formattedTrivyReportFilename}.*", allowEmptyArchive: true
```

Comment:
In this method, `trivyDirectory` already holds the slash (line 109): `String trivyDirectory = "trivy/"`

Reply:
We should not hard code this directory. Please use the field of the trivy class for the working directory. then @meiserloh s comment is valid again.

Reply:
Which field are you referring to? `script.env.HOME`?
And why shouldn't we hardcode it? The user has no contact with that directory, because it only temporarily stores the converted files. These files are then archived to the Jenkins build/job and can be viewed/downloaded by the user.

Reply:
Done, see https://github.com/cloudogu/ces-build-lib/commit/12c2a304bb73da8303dd04ae70ac57ef45ce6a57 and https://github.com/cloudogu/ces-build-lib/commit/69b92099acf57764f69a96460ffc9c16405b15de